From 4937de68528ae0f3ee3504056144d7cffba6fa24 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Wed, 31 Jan 2024 14:39:24 +0100
Subject: [PATCH 0715/1695] .gitignore: Ignore all dot files and folders except
 the standard ones
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Improve the gitignore to handle dot files and dot folders that are
created by most of development tools, such as IDEs.

This patch allows OPTEE-OS developers to store their specific tool
configurations under dot files or folders and not be bothered by the output
of the git status command.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f10d4a37a18de1bc49ee1a5d2b252660df684b00 .gitignore: Ignore all dot files and folders except the standard ones]
Change-Id: I8421c13efde9f7cdd5c0928af04e66df7cec17f8
Signed-off-by: Clément Le Goffic <clement.legoffic@foss.st.com>
(cherry picked from commit 40944c5ccf0b3bcb0adbfbd6e3a40b928a2a250f)
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/371643
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .gitignore | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/.gitignore b/.gitignore
index c7a0bbe38..79de5e9f7 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,8 +1,14 @@
-/.cproject
 /compile_commands.json
 /cscope.*
 /tags
 /TAGS
 /out
 /.project
-*.swp
+# Ignore all dot files and folder
+.*
+# But not those
+!.clang-format
+!.checkpatch.conf
+!.gitattributes
+!.gitignore
+!.github
-- 
2.47.2

